### PR TITLE
fix(BrowserAdapter): correctly removes styles on IE

### LINKED ISSUE #7916: Styles not being removed when evaluated to null in IE
**Steps to reproduce and a minimal demo of the problem**

See this plnkr http://plnkr.co/edit/WPKg2KgIXMIxugSpXEMQ?p=preview

**What steps should we try in your demo to see the problem?**

Press the "change my bg color" button as much as you want, it should toggle the div's background from none to red, and viceversa.

Now, to see it working for real in IE, change 

```
'[style.background-color]' : `isRed ? 'red': null`
```
to
```
'[style.background-color]' : `isRed ? 'red': ''`  // change 'null' to single quotes
```
with that will work correctly in both chrome, IE and FF. 

I got this solution/workaround from @alexhub in gitter's chatroom.

**Current behavior**

In chrome works perfectly, but in IE the background color is added but not removed.

**Expected/desired behavior**

It should work with `null` in all browsers, or it should be changed to `''` to work with most of them.

**Other information**

IE11  - 11.0.9600.18230
IE10 (emulated)
IE9 (emulated)

All with the same behavior. 

Should [this line](https://github.com/angular/angular/blob/master/modules/angular2/src/platform/browser/browser_adapter.ts#L211) be changed to use an empty string instead?

```
element.style[stylename] = '';
```

### PR DESCRIPTION
**Pull request by vicb**
fixes #7916

### CODE REVIEW

**Comment by vicb** [comment]
From [jQuery docs](http://api.jquery.com/css/#css2):

```
Setting the value of a style property to an empty string — e.g. $( "#mydiv" ).css( "color", "" ) — removes that property from an element if it has already been directly applied, whether in the HTML style attribute, through jQuery's .css() method, or through direct DOM manipulation of the style property. As a consequence, the element's style for that property will be restored to whatever value was applied. So, this method can be used to cancel any style modification you have previously performed. It does not, however, remove a style that has been applied with a CSS rule in a stylesheet or <style> element. Warning: one notable exception is that, for IE 8 and below, removing a shorthand property such as border or background will remove that style entirely from the element, regardless of what is set in a stylesheet or <style> element
```

**Comment by chuckjaz** [comment]
Please resolve merge conflict.

**Comment by pkozlowski-opensource** [comment]
@vicb is there a way of adding a unit test for this? I'm worried that this issue might come back in this or another form if we don't have a test. And since we are running all tests in IE this could be verified automatically.

**Comment by vicb** [comment]
@pkozlowski-opensource please create an issue an I'll look at it (assign it to me). thanks.

@chuckjaz rebased

**Review by chuckjaz — CHANGES_REQUESTED** [review]

File: modules/@angular/platform-browser/src/browser/browser_adapter.ts [hunk 1]
```diff
@@ -71,67 +71,45 @@ var _chromeNumKeyPadMap = {
 export class BrowserDomAdapter extends GenericBrowserDomAdapter {
   parse(templateHtml: string) { throw new Error('parse not implemented'); }
   static makeCurrent() { setRootDomAdapter(new BrowserDomAdapter()); }
-  hasProperty(element: any /** TODO #9100 */, name: string): boolean { return name in element; }
-  setProperty(el: /*element*/ any, name: string, value: any) { el[name] = value; }
-  getProperty(el: /*element*/ any, name: string): any { return el[name]; }
-  invoke(el: /*element*/ any, methodName: string, args: any[]): any {
-    el[methodName].apply(el, args);
-  }
+  hasProperty(element: Node, name: string): boolean { return name in element; }
```
Comment (vicb) [inline modules/@angular/platform-browser/src/browser/browser_adapter.ts]: no part of the public API

**Review by chuckjaz — APPROVED** [review]

**Pull request merged**